junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/discovery/MethodSelectorResolver.java

```diff
@@ -97,14 +97,15 @@ private Resolution resolve(Context context, List<Class<?>> enclosingClasses, Cla
 				.collect(toSet());
 		// @formatter:on
 		if (matches.size() > 1) {
-			logger.warn(() -> {
-				Stream<TestDescriptor> testDescriptors = matches.stream().map(Match::getTestDescriptor);
-				return String.format(
-					"Possible configuration error: method [%s] resulted in multiple TestDescriptors %s. "
-							+ "This is typically the result of annotating a method with multiple competing annotations "
-							+ "such as @Test, @RepeatedTest, @ParameterizedTest, @TestFactory, etc.",
-					method.toGenericString(), testDescriptors.map(d -> d.getClass().getName()).collect(toList()));
-			});
+			Stream<TestDescriptor> testDescriptors = matches.stream().map(Match::getTestDescriptor);
+			String message = String.format(
+				"Possible configuration error: method [%s] resulted in multiple TestDescriptors %s. " //
+						+ "This is typically the result of annotating a method with multiple competing annotations " //
+						+ "such as @Test, @RepeatedTest, @ParameterizedTest, @TestFactory, etc.",
+				method.toGenericString(), testDescriptors.map(d -> d.getClass().getName()).collect(toList()));
+			context.reportIssue(EngineDiscoveryIssue.builder(Severity.NOTICE, message) //
+					.selector(selector) //
+					.source(MethodSource.from(testClass, method)));
```

Comment:
:speech_balloon: Engines using the `EngineDiscoveryRequestResolver` framework work discovery can report issues via the `Context` passed to each `SelectorResolver`.